lib/rmt/cli/decorators/system_decorator.rb

```diff
@@ -1,15 +1,29 @@
 class RMT::CLI::Decorators::SystemDecorator < RMT::CLI::Decorators::Base
 
-  def initialize(systems)
-    @data = systems.map do |system|
-      [
-        system.login,
-        system.hostname,
-        system.registered_at,
-        system.last_seen_at,
-        products_and_subscriptions(system).join("\n")
-      ]
-    end
+  attr_reader :data
+
+  def initialize(systems, all: false)
+    @data = if all
+              [
+                [
+                  systems.login,
+                  systems.hostname,
+                  systems.registered_at,
+                  systems.last_seen_at,
+                  products_and_subscriptions(systems).join("\n")
+                ]
+              ]
```

Comment:
As I commented before, maybe if we use batches here instead of a single instance, we can eliminate this conditional and the `all` flag.

Looking at the class alone, it's not clear why the `all` flag, when it's `true`, treat the `systems` variable as a single instance.